launcher/src/backend/NodeUpdates.js

```diff
@@ -19,7 +19,7 @@ export class NodeUpdates {
       await this.nodeConnection.runPlaybook("upgrade-prep", { stereum_role: "upgrade-prep" });
       log.info("Starting Upgrade to noble numbat ...");
       const result = await this.nodeConnection.sshService.exec(
-        "do-release-upgrade -d -f DistUpgradeViewNonInteractive --allow-third-party"
+        "do-release-upgrade -f DistUpgradeViewNonInteractive --allow-third-party && shutdown now -r"
       );
       await this.nodeConnection.restartServer();
```

Comment:
you'll need to adapt here:

- You can `touch /var/run/reboot-required` insead of `shutdown now -r`
- You can adapt the restartServer method so , that you pass a tag something like `forceReboot` so it skips the check for reboot required and runs the reboot anyway.